guides/developer/service-overviews/orca.md

```diff
@@ -0,0 +1,48 @@
+
+A Message is a fine-grained unit of work. As an example, a `StartStage` message will verify that all upstream branches are complete and then perform any logic associated with preparing for the stage's execution, it won't actually perform the stage's core actions. A `RunTask` is really the only message that actually performs stage work that a user would be familiar with.
+
+Some message types are redeliverable and duplicated, whereas others are not. For example, in a pipeline that has two concurrent stage branches that do not join at the end, there will be two `CompleteExecution` messages sent. This is fine, because the `CompleteExecutionHandler` has logic to verify that indeed all stages are in a completed state before marking the entire execution as complete, whether it ultimately be a failure or success.
+
+For every message, there is a single correlated `Handler` that contains all of the logic for processing this `Message`. A Message's contents are small and the bare minimum of information to process the request: For instance, an Execution is referenced by ID rather than containing its entire state. Orca builds atop these foundational Handlers in the form of `StageDefinitionBuilder` and other classes to build the DAG and implement actual Spinnaker orchestration logic.
+
+The queue can be categorized into two parts: The **QueueProcessor**, and its Handlers. Orca uses a single thread to run `QueueProcessor`, which polls the oldest messages "ready" off the queue. A ready message is one whose delivery time is now or in the past: It is normal to have many more messages in the queue than those that are actually ready. 
+
+A separate worker thread pool is used for Handlers. Orca’s ability to scale to process its work is dependent on availability of threads in the worker pool which can be tuned either by adjusting the worker thread pool size, or by increasing the number of Orca instances: Ignoring downstream bottlenecks, Orca can be scaled out horizontally to meet the work demands Spinnaker is given.
+
+If the queue starts to back up, Executions and their Tasks will begin to take longer to start or complete. Sometimes the queue will back up because of downstream service pressure (for example, Clouddriver), but may also be due to Orca not having enough threads or persistence capacity.
+
+One further note about the `QueueProcessor` and its thread pool. The QueueProcessor will poll the queue on a regular interval (default 10ms) and will attempt to fill its thread pool if there's work ready to be processed. If the thread pool has a core size of 20, and 5 are already busy, it will attempt to take 15 messages off the queue immediately rather than one. This is because in most cases a message will take less than 1ms to complete and we don't want Orca to idle unnecessarily.
```

Comment:
s/will poll/polls/
s/will attempt/tries/
s/be processed/process/
s/will take/takes/